Frontend display for claim interval & amount. Ref #11 #15

darwintree opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19)
<!-- Reviewable:end -->

Commit:
Frontend will show real-time token claim interval and amount. CFX can now be configured independent of default val. Ref #11 #15

Commit:
Fix a bug for claim

wangdayong228 commented on frontend/src/contracts-config/index.js:
```diff
@@ -6,18 +6,26 @@ import { default as faucet } from "../../../build/contracts/Faucet.json";
 const options = [
+  {
+    // 这个地址是测试网的零地址，特殊地，在水龙头合约中该地址会被视为CFX的地址
+    // 注意到在dapp中不存在任何操作会将代币合约地址作为 to，因此不会产生发向零地址的交易，使用这个地址是安全的
```
dapp可能会有向0地址转账的操作，一般用于销毁代币。但对我们这里没有影响，因为0地址一定不是一个合约地址。

wangdayong228 reviewed (changes requested):
> Reviewed 7 of 7 files at r1.
> *[Reviewable](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19)* status: all files reviewed, 2 unresolved discussions (waiting on @darwintree and @wangdayong228)

___
*[contracts/Faucet.sol, line 135 at r1](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbK8FTsCGydd1yZPy9A:-MbK8FTt3s5yFL5YMZRo:bf6bovq) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/2679cd8a733622c02816ebe516934a4e3b25ad49/contracts/Faucet.sol#L135)):*
> ```Solidity
>     }
> 
>     function claimCfx() public payable {
> ```

合约还可以优化一下
- claimToken 改为 claim；业务逻辑包含claimCfx的，只是最后的部分获取余额及转账方式不同
- calimCfx 只需要调用claimToken即可
- claimTokenRegardingSenderBalance 同理


<!-- Sent from Reviewable.io -->

Commit:
优化合约逻辑 并添加部分注释。
关闭了Production模式下余额更新的相关代码
移除了部分模块中没有用到的代码

Commit:
调整 contracts-config 文件结构.添加部分注释

Commit:
移除了与locale相关的历史遗留代码. 移除了不必要的网络环境tooltip

Commit:
给出Conflux Portal加载时的错误提示.
将 networkVersion（deprecated） 替换为 chainId

Commit:
将 isNativeToken 变为计算属性
调整报错等对话框样式（优化了换行）
调整提醒文本

Commit:
Remove web3

Commit:
移除冗余代码

Commit:
Deploy new contract

Commit:
Update README

darwintree reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19)* status: 1 of 19 files reviewed, 2 unresolved discussions (waiting on @wangdayong228)

___
*[contracts/Faucet.sol, line 135 at r1](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbK8FTsCGydd1yZPy9A:-MbKcG5U5VloZ370LVWN:b91b9wy) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/2679cd8a733622c02816ebe516934a4e3b25ad49/contracts/Faucet.sol#L135)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

合约还可以优化一下
- claimToken 改为 claim；业务逻辑包含claimCfx的，只是最后的部分获取余额及转账方式不同
- calimCfx 只需要调用claimToken即可
- claimTokenRegardingSenderBalance 同理


</blockquote></details>

done. 
claimToken 接口命名不变，但是传入token地址为0时会调用cfx的相关逻辑。
claimCfx接口仍然保留，但是前端不再调用此接口了。
claimTokenRegardingSenderBalance也进行了对应的修改
___
*[frontend/src/contracts-config/index.js, line 11 at r1](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbK8BfH3ebzjOnZORKS-r1-11:-MbQbnVVAoyD6lRvgrqd:b2fjvr9) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/2679cd8a733622c02816ebe516934a4e3b25ad49/frontend/src/contracts-config/index.js#L11)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

dapp可能会有向0地址转账的操作，一般用于销毁代币。但对我们这里没有影响，因为0地址一定不是一个合约地址。
</blockquote></details>

done. 修正了一下注释


<!-- Sent from Reviewable.io -->

wangdayong228 reviewed (changes requested):
> Reviewed 18 of 18 files at r2.
> *[Reviewable](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19)* status: all files reviewed, 5 unresolved discussions (waiting on @wangdayong228)

___
*[contracts/Faucet.sol, line 76 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZNWBz1i_HWRlcwdQC:-MbZNWC-6ZSnZuIhYjbr:boxurmv) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L76)):*
> ```Solidity
>     @param decimals 设置每次领取 Token 的数位，与 amountForDecimals 一起使用，领取额度为 amountForDecimals * (10^decimals)
>      */
>     function setClaimSetting(address tokenContractAddress, uint256 interval, uint256 amountForDecimals, uint256 decimals) public onlyManager {
> ```

decimals如果为cfx则18，如果为token直接从合约中读取，避免手动设置导致错误
___
*[frontend/src/components/FaucetPanel.vue, line 382 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZOhvl1P3iwI3OipV8:-MbZOhvm87m_j2Sm-We_:batgt4m) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/frontend/src/components/FaucetPanel.vue#L382)):*
> ```Vue
>       if (newVal) {
>         // 异步操作 不使用await
>         this.updateFaucetCfxBalance();
> ```

这里可以使用Promise.all同时处理多个异步
___
*[frontend/src/components/FaucetPanel.vue, line 393 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZOqMK7lOHC0BCLBnX:-MbZOqMK7lOHC0BCLBnY:batgt4m) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/frontend/src/components/FaucetPanel.vue#L393)):*
> ```Vue
>         // 不过相应的 也无法抛出错误了
>         this.updateFaucetInterval();
>         this.updateFaucetAmount();
> ```

这里可以使用Promise.all同时处理多个异步
___
*[frontend/src/components/FaucetPanel.vue, line 403 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZOsHsAeE-PUSNlGle:-MbZOsHsAeE-PUSNlGlf:batgt4m) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/frontend/src/components/FaucetPanel.vue#L403)):*
> ```Vue
>         // 不过相应的 也无法抛出错误了
>         this.updateFaucetInterval();
>         this.updateFaucetAmount();
> ```

这里可以使用Promise.all同时处理多个异步


<!-- Sent from Reviewable.io -->

wangdayong228 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19)* status: all files reviewed, 8 unresolved discussions (waiting on @wangdayong228)

___
*[contracts/Faucet.sol, line 66 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZ__Oq6JCdybZ5KYQB:-MbZ__Or9gcH1jKEDtjA:b-wohups) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L66)):*
> ```Solidity
>     }
> 
>     function setDefaultAmount(uint256 cfxAmount) public onlyManager {
> ```

cfxAmount 改为 amount, 不乘 1e18, 具体的值让开发者设置
___
*[contracts/Faucet.sol, line 76 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZaiBXDfNQFiFqU4XZ:-MbZaiBYCoNMHODi60su:b-vw9a84) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L76)):*
> ```Solidity
>     @param decimals 设置每次领取 Token 的数位，与 amountForDecimals 一起使用，领取额度为 amountForDecimals * (10^decimals)
>      */
>     function setClaimSetting(address tokenContractAddress, uint256 interval, uint256 amountForDecimals, uint256 decimals) public onlyManager {
> ```

amountForDecimals改为amount，该值就用实际值，不帮用户处理精度。去掉参数decimals
___
*[contracts/Faucet.sol, line 78 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZafP23h1e-G7Kq47k:-MbZafP23h1e-G7Kq47l:b-me393r) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L78)):*
> ```Solidity
>     function setClaimSetting(address tokenContractAddress, uint256 interval, uint256 amountForDecimals, uint256 decimals) public onlyManager {
>         tokenClaimSettings[tokenContractAddress].interval = interval;
>         tokenClaimSettings[tokenContractAddress].amount = amountForDecimals * (10**decimals);
> ```

不用乘 10**decimals
___
*[contracts/Faucet.sol, line 82 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZcWuz6uRwnTnvRxOQ:-MbZcWv-6I_szNTkCK1y:b-kdibxv) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L82)):*
> ```Solidity
> 
>     function nextCfxClaim() public view returns (uint256) {
>         return nextTokenClaim(address(0));
> ```

这个方法没有看到有用的地方，如果没有用可以去掉
___
*[contracts/Faucet.sol, line 85 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZbJy8Exkvm46pzPYZ:-MbZbJy8Exkvm46pzPY_:b-kdibxv) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L85)):*
> ```Solidity
>     }
> 
>     function nextTokenClaim(address tokenContractAddress)
> ```

这个方法没有看到有用的地方，如果没有用可以去掉


<!-- Sent from Reviewable.io -->

Commit:
Use Promise.all

Commit:
Change page title. ref #21.
Manager can retrieve token from faucet contract. ref #20
Add update log in readme

Commit:
Remove contract bytecode import

Commit:
Release v1.0.0

Commit:
修复CLAIM按钮上不显示提示的bug

darwintree reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19)* status: all files reviewed, 8 unresolved discussions (waiting on @wangdayong228)

___
*[contracts/Faucet.sol, line 66 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZ__Oq6JCdybZ5KYQB:-MbabtUx2ivsS0uwCk7o:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L66)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

cfxAmount 改为 amount, 不乘 1e18, 具体的值让开发者设置
</blockquote></details>

done
___
*[contracts/Faucet.sol, line 76 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZaiBXDfNQFiFqU4XZ:-Mbabm4RD3z4EFQB9lJt:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L76)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

amountForDecimals改为amount，该值就用实际值，不帮用户处理精度。去掉参数decimals
</blockquote></details>

done
___
*[contracts/Faucet.sol, line 76 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZNWBz1i_HWRlcwdQC:-MbabuPX95HwPrca5jX5:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L76)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

decimals如果为cfx则18，如果为token直接从合约中读取，避免手动设置导致错误
</blockquote></details>

done
___
*[contracts/Faucet.sol, line 78 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZafP23h1e-G7Kq47k:-MbaboGg-ydlBA8lZHVw:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L78)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

不用乘 10**decimals
</blockquote></details>

done
___
*[contracts/Faucet.sol, line 82 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZcWuz6uRwnTnvRxOQ:-Mbabozk6c61k3u9gR-6:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L82)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

这个方法没有看到有用的地方，如果没有用可以去掉
</blockquote></details>

done
___
*[contracts/Faucet.sol, line 85 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZbJy8Exkvm46pzPYZ:-MbabpXD49uoOXXyUp09:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/contracts/Faucet.sol#L85)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

这个方法没有看到有用的地方，如果没有用可以去掉
</blockquote></details>

done
___
*[frontend/src/components/FaucetPanel.vue, line 382 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZOhvl1P3iwI3OipV8:-MbabqPO9gjudfnyfHEn:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/frontend/src/components/FaucetPanel.vue#L382)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

这里可以使用Promise.all同时处理多个异步
</blockquote></details>

done
___
*[frontend/src/components/FaucetPanel.vue, line 393 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZOqMK7lOHC0BCLBnX:-Mbabr3q0-Fe3P7Rz0iL:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/frontend/src/components/FaucetPanel.vue#L393)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

这里可以使用Promise.all同时处理多个异步
</blockquote></details>

done
___
*[frontend/src/components/FaucetPanel.vue, line 403 at r2](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19#-MbZOsHsAeE-PUSNlGle:-MbabrfQ3UBCZBO2X-5E:bb7f2mf) ([raw file](https://github.com/conflux-fans/conflux-faucet/blob/c6687d307e2ebf7c40ca77ba5468b15c1a57ae3e/frontend/src/components/FaucetPanel.vue#L403)):*
<details><summary><i>Previously, wangdayong228 wrote…</i></summary><blockquote>

这里可以使用Promise.all同时处理多个异步
</blockquote></details>

done


<!-- Sent from Reviewable.io -->

darwintree committed:
Remove redundant code

wangdayong228 reviewed (approved):
> Reviewed 11 of 12 files at r3, 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/conflux-fans/conflux-faucet/19)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @darwintree)



<!-- Sent from Reviewable.io -->
